core/arch/arm/kernel/boot.c

```diff
@@ -1388,14 +1388,18 @@ void __weak boot_save_args(unsigned long a0, unsigned long a1,
 	 * a4	- Not used
 	 */
 
-	if (IS_ENABLED(CFG_TRANSFER_LIST) &&
-	    a1 == (TRANSFER_LIST_SIGNATURE | REG_CONVENTION_VER_MASK)) {
-		if (IS_ENABLED(CFG_ARM64_core)) {
+	if (IS_ENABLED(CFG_TRANSFER_LIST)) {
+		if (IS_ENABLED(CFG_ARM64_core) &&
+			a1 == TRANSFER_LIST_HANDOFF_X1_VALUE(REG_CONVENTION_VER)) {
 			boot_save_transfer_list(a2, a3, a0);
 			boot_arg_fdt = a0;
-		} else {
+		} else if (
+			a1 == TRANSFER_LIST_HANDOFF_R1_VALUE(REG_CONVENTION_VER)) {
 			boot_save_transfer_list(a0, a3, a2);
 			boot_arg_fdt = a2;
+		} else {
+			EMSG("Wrong TL register convention...\n");
```

Comment:
How about `panic("Invalid TL register convention");`?
Is the intent that if `CFG_TRANSFER_LIST` is enabled we'll either save an OK transfer list or panic?

Reply:
To be honest, I'm not sure It could progress futher If transfer_list doesn't deliver properly whne `CFG_TRANSFER_LIST` enabled.

But since former code allows to progress futher in this situation, I'll remove panic :) 

Thanks to let me know.